src/Nethermind/Ethereum.Test.Base/BlockchainTestBase.cs

```diff
@@ -247,6 +204,13 @@ protected async Task<EthereumTestResult> RunTest(BlockchainTest test, Stopwatch?
         await blockchainProcessor.StopAsync(true);
         stopwatch?.Stop();
 
+        IBlockCachePreWarmer? preWarmer = container.Resolve<MainBlockProcessingContext>().LifetimeScope.ResolveOptional<IBlockCachePreWarmer>();
```

Comment:
Not sure if, but does this create any issue with parallel test invocation? If not can you be more specific about it? I think maybe `LifetimeScope` solves this but it isn't totally clear for me.

Reply:
It does not? Each invocation creates a completely different instance.